core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -126,6 +128,11 @@ public Environment(Settings settings) {
             dataFiles = new Path[]{homeFile.resolve("data")};
             dataWithClusterFiles = new Path[]{homeFile.resolve("data").resolve(ClusterName.clusterNameFromSettings(settings).value())};
         }
+        if (settings.get("path.shared_data") != null) {
+            sharedDataFile = PathUtils.get(cleanPath(settings.get("path.shared_data")));
+        } else {
+            sharedDataFile = homeFile.resolve("data");
```

Comment:
I don't want to set it as null because if we remove `custom_paths_enabled` we'll now get a NullPointerException if someone tries to use it. I think if we remove `custom_paths_enabled` we can set this to a better default if you're concerned about it using the data path. Maybe just `${es.path.home}/data/custom` or something?